test/bdd/agent/agent_controller_steps.go

```diff
@@ -87,16 +109,44 @@ func (a *ControllerSteps) checkAgentIsRunning(
 	return nil
 }
 
-func (a *ControllerSteps) healthCheck(url string) error {
-	resp, err := http.Get(url) //nolint: gosec
-	if err != nil {
+func (a *ControllerSteps) checkAgentIsRunningWithHTTPInbound(agentID, inboundHost,
+	inboundPort, controllerURL, webhookURL string) error {
+	if err := a.checkAgentIsRunning(agentID, controllerURL, webhookURL); err != nil {
 		return err
 	}
 
-	err = resp.Body.Close()
-	if err != nil {
-		logger.Errorf("Failed to close response body : %s", err)
+	// verify inbound
+	if err := a.healthCheck(fmt.Sprintf("http://%s:%s", inboundHost, inboundPort)); err != nil {
+		logger.Debugf("Unable to reach inbound '%s' for agent '%s', cause : %s", controllerURL, agentID, err)
+		return err
 	}
 
+	logger.Debugf("Agent '%s' running inbound on '%s' and port '%s'", agentID, inboundHost, inboundPort)
+
 	return nil
 }
+
+func (a *ControllerSteps) healthCheck(url string) error {
+	if strings.HasPrefix(url, "http") {
+		resp, err := http.Get(url) //nolint: gosec
+		if err != nil {
+			return err
+		}
+
+		err = resp.Body.Close()
+		if err != nil {
+			logger.Errorf("Failed to close response body : %s", err)
+		}
+
+		return nil
+	} else if strings.HasPrefix(url, "ws") {
+		_, _, err := websocket.Dial(goctx.Background(), url, nil) // nolint bodyclose
```

Comment:
why isn't the body being closed?

Reply:
Missed to remove this nolint. [Updated](https://github.com/hyperledger/aries-framework-go/compare/65a3f4ae6435f3be1caa9b23918326908c16c298..4bc13cf3b591897f05d48276eb36fdb6bc73a0f8) the code now.